internal/service/domain/sensor/sensor.go

```diff
@@ -166,7 +165,10 @@ func (s *SensorService) MapSensorToTree(ctx context.Context, sen *entities.Senso
 	}
 
 	if nearestTree != nil {
-		_, err = s.treeRepo.Update(ctx, nearestTree.ID, tree.WithSensor(sen))
+		_, err = s.treeRepo.Update(ctx, nearestTree.ID, func(tree *entities.Tree) (bool, error) {
+			tree.Sensor = sen
+			return true, nil
+		})
```

Comment:
Can you add a debug log that the sensor has been updated, something like this:
```suggestion
		_, err = s.treeRepo.Update(ctx, nearestTree.ID, func(tree *entities.Tree) (bool, error) {
			tree.Sensor = sen
			log.Debug("update sensor on tree", "tree_id", tree.ID, "sensor_id", sen.ID)
			return true, nil
		})
```